src/modules/mavlink/mavlink_mission.h

```diff
@@ -136,8 +137,11 @@ class MavlinkMissionManager
 
 	static bool		_transfer_in_progress;			///< Global variable checking for current transmission
 
-	uORB::Subscription	_mission_result_sub{ORB_ID(mission_result)};
 	uORB::SubscriptionData<mission_s> 	_mission_sub{ORB_ID(mission)};
+	uORB::SubscriptionData<mission_result_s> _mission_result_data{ORB_ID(mission_result)};
+	uORB::Subscription _vehicle_status_sub{ORB_ID(vehicle_status)};
+
+	vehicle_status_s _vehicle_status;
```

Comment:
Instead of storing the full vehicle_status message in a member variable, how about storing `nav_state` (as `_nav_state`)?